File: Store-Shoe-Online/Store-Shoe-Online/ViewModel/UserInfoVM.cs
using Store_Shoe_Online.Data;
using System.ComponentModel.DataAnnotations;

namespace Store_Shoe_Online.ViewModel
{
    public class UserInfoVM
    {
        public string? UserId { get; set; }
        public string? UserName { get; set; }
        public string? Email { get; set; }
        [Phone]
        public string? NumberPhone { get; set; }
        public string? FullName { get; set; }
        public string? Password { get; set; }
        public string? PasswordOld { get; set; }

        public UserInfoVM() { }

        public UserInfoVM(ApplicationUser user)
        {
            this.FullName = user.FullName;
            this.UserName = user.UserName;
            this.Email = user.Email;
            this.NumberPhone = user.PhoneNumber;
            this.UserId = user.Id;
        }
    }
}

File: App Mobile/flutter_store_shoe/lib/models/UserInfoVM.dart
// ignore_for_file: file_names

import 'package:shared_preferences/shared_preferences.dart';

class UserInfoVM {
  String? userId = "";
  String? userName;
  String? email;
  String? numberPhone;
  String? fullName;
  String? password;
  String? passwordOld;

  UserInfoVM({
    this.userId,
    this.userName,
    this.email,
    this.numberPhone,
    this.fullName,
    this.password,
    this.passwordOld,
  });

  factory UserInfoVM.fromJson(Map<String, dynamic> json) {
    return UserInfoVM(
      userId: json['userId'] ?? '',
      userName: json['userName'] ?? '',
      email: json['email'] ?? '',
      numberPhone: json['numberPhone'] ?? '',
      fullName: json['fullName'] ?? '',
      password: json['password'] ?? '',
      passwordOld: json['passwordOld'] ?? '',
    );
  }

  Future<void> saveToPrefs() async {
    final prefs = await SharedPreferences.getInstance();
    await prefs.setString('userId', userId ?? '');
    await prefs.setString('userName', userName ?? '');
    await prefs.setString('email', email ?? '');
    await prefs.setString('numberPhone', numberPhone ?? '');
    await prefs.setString('fullName', fullName ?? '');
    await prefs.setString('password', password ?? '');
    await prefs.setString('passwordOld', passwordOld ?? '');
  }

  // Tạo một đối tượng UserInfoVM từ SharedPreferences
  factory UserInfoVM.fromPrefs(SharedPreferences prefs) {
    return UserInfoVM(
      userId: prefs.getString('userId'),
      userName: prefs.getString('userName'),
      email: prefs.getString('email'),
      numberPhone: prefs.getString('numberPhone'),
      fullName: prefs.getString('fullName'),
      password: prefs.getString('password'),
      passwordOld: prefs.getString('passwordOld'),
    );
  }
}
